test/integration/targets/prepare_nxos_tests/tasks/main.yml

```diff
@@ -35,7 +35,19 @@
   connection: network_cli
   register: intout
 
-- set_fact: intdataraw="{{ intout.stdout_lines[0]['TABLE_interface']['ROW_interface'] }}"
+- block:
+  - set_fact: intdataraw="{{ intout.stdout_lines[0]['TABLE_interface']['ROW_interface'] }}"
+
+  - name: "Remove logical interfaces (sub-ints, loopbacks, port-channels, etc)"
+    cli_config:
+      config: |
+        no interface {{ item.interface }}
+    connection: network_cli
+    loop: "{{ intdataraw }}"
+    when: item.interface is regex('port-channel|loopback|\.[0-9]+$')
+    ignore_errors: yes
+  no_log: True
+
```

Comment:
So, for this. I'd like to see each test clean up after isn't self, rather then prepare. The reason for this, doing it here, isn't really efficient. Today, we likely spend on about 1.5 hours just doing prepare_nxos_tests things. In the case of zuul testing I am working on, we don't actually run this code but create the variables in the inventory file.

Reply:
@pabelanger I'll remove this. It was a convenient solution for us since we share our testbeds with more than just ansible testing and they're often left in a stale state.

Reply:
thanks! in our case, our nxos instances are ephemeral, so between different PRs, we are safe from it.  I believe, we can many work to update your test-bed to do this before running ansible-test, and update tests to try and clean up leaked data.

Reply:
It's been removed. I'll add this cleanup to our local CI infra, that should cover the majority of cases. Agreed, we should also keep an eye out for dirty test teardowns.

Reply:
Hmm, do you mind removing the whitespace here too? Trying to get zuul to work, and this is a merge conflict for us